src/io/flutter/run/LaunchState.java

```diff
@@ -326,6 +328,21 @@ protected final RunContentDescriptor doExecute(@NotNull RunProfileState state, @
 
         if (app != null && StringUtil.equals(app.deviceId(), selectedDeviceId)) {
           if (executorId.equals(app.getMode().mode())) {
+
+            if (!identicalCommands(app.getCommand(), launchState.runConfig.getCommand(env, app.device()))) {
+              // To be safe, relaunch as the arguments to launch have changed.
+              try {
+                // TODO(jacobr): ideally we shouldn't be synchronously waiting
```

Comment:
Do we expect people to change launch config options and then do a full restart? I would expect that the new options wouldn't be applied until the next full app launch.

Reply:
this isn't the "full restart" case.
This is if the user clicks on "Debug main.dart"
after changing the config.
Currently even though the config changes, surprisingly all they get is a "full restart".
If the user triggers an actual "full restart" with this change, they still get a "full restart" even though the config has changed.

Reply:
OK, it sounds like this is here to reduce surprise where the cli launch args change, the user hits the run or debug buttons, but we recycle an existing launch with different launch args.

👍